apps/backend-session/src/lib/xion/backend/AbstraxionBackend.ts

```diff
@@ -0,0 +1,478 @@
+import { randomBytes } from "node:crypto";
+import NodeCache from "node-cache";
+import type { IncomingMessage } from "node:http";
+import { GasPrice } from "@cosmjs/stargate";
+import {
+  AbstraxionAuth,
+  ContractGrantDescription,
+  SpendLimit,
+} from "@burnt-labs/abstraxion-core";
+import { fetchConfig, xionGasValues } from "@burnt-labs/constants";
+import {
+  AbstraxionBackendConfig,
+  ConnectionInitResponse,
+  CallbackRequest,
+  CallbackResponse,
+  StatusResponse,
+  DisconnectResponse,
+  Permissions,
+  XionKeypair,
+} from "./types";
+import * as e from "./types/errors";
+import { SessionKeyManager } from "./services/SessionKeyManager";
+import {
+  DatabaseRedirectStrategy,
+  DatabaseStorageStrategy,
+  InMemoryDummyRedirectStrategy,
+} from "./adapters/AbstraxionStategies";
+
+export class AbstraxionBackend {
+  public readonly sessionKeyManager: SessionKeyManager;
+  private readonly _stateStore: NodeCache;
+  private readonly _gasPriceDefault: GasPrice;
+
+  constructor(private readonly config: AbstraxionBackendConfig) {
+    // Validate configuration
+    if (!config.encryptionKey) {
+      throw new e.EncryptionKeyRequiredError();
+    }
+    if (!config.databaseAdapter) {
+      throw new e.DatabaseAdapterRequiredError();
+    }
+    if (!config.redirectUrl) {
+      throw new e.RedirectUrlRequiredError();
+    }
+    if (!config.treasury) {
+      throw new e.TreasuryRequiredError();
+    }
+    if (!config.rpcUrl) {
+      throw new e.RpcUrlRequiredError();
+    }
+
+    let gasPriceDefault: GasPrice;
```

Comment:
We can just use the value in the constant package here. No need to check the network.